flaps/flaps.go

```diff
@@ -198,10 +202,42 @@ func (f *Client) _sendRequest(ctx context.Context, method, endpoint string, in,
 	}
 	if out != nil {
 		if err := json.NewDecoder(resp.Body).Decode(out); err != nil {
-			return err
+			return resp.StatusCode, err
+		}
+	}
+	return resp.StatusCode, nil
+}
+
+type flapsCall struct {
+	Call       string  `json:"c"`
+	Command    string  `json:"co"`
+	StatusCode int     `json:"s"`
+	Duration   float64 `json:"d"`
+}
+
+func sendFlapsCallMetric(ctx context.Context, endpoint flapsAction, statusCode int, duration time.Duration) {
+	cmdCtx := command_context.FromContext(ctx)
+
+	// Iterate backwards through the command name to figure out the command being run.
+	// For example, `fly m run` becomes `machine-run`, `deploy --flags` becomes `deploy`
+	// TODO(billy): Unit test this
+	var nameParts []string
+	for cmdCtx != nil {
+		// Don't include the binary name in the nameParts
+		if cmdCtx.Parent() == nil {
+			break
```

Comment:
nit: it can be part of the `for` condition like in:
```go
for cmdCtx != nil && cmdCtx.Parent() != nil {
```

Reply:
go is too advanced for my small brain